x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/dimension_container.tsx

```diff
@@ -49,8 +51,16 @@ export function DimensionContainer({
     }
   }, [isOpen]);
 
+  const closeOnEscape = (event: KeyboardEvent) => {
```

Comment:
Nit: Can we `useCallback`-ify this? Looking at the `EuiWindowEvent` code, it will unregister/register the listener every time otherwise: https://github.com/elastic/eui/blob/master/src/services/window_event/window_event.ts#L34

Reply:
We could. I'm just worried the overuse of `useCallback` in general (aka [dies by a thousands of cuts](https://royi-codes.vercel.app/thousand-usecallbacks/)) and found no particular bottleneck in this case: the function is performing a very light task and the only upside is the stability of the function reference.

In the EUI Flyout component the callback is not memoized either: https://github.com/elastic/eui/blob/master/src/components/flyout/flyout.tsx#L112

If strictly required for performance I'll change it.

Reply:
It's definitely not a big issue, it just stood out to me. I think this specific case is point 4 on the "do" list for memoizing in the blog article you linked:
> Do useCallback/useMemo to prevent repeatedly triggering useEffect() in a child (ie: if the useEffect has the function or object as a dependency, changing its reference could repeatedly tear down/create event listeners)